tests/openassetio-core-c/private/StringViewTest.cpp

```diff
@@ -38,6 +41,49 @@ SCENARIO("Creating, modifying and querying a C API mutable StringView") {
         THEN("view has been updated") { CHECK(actualStringView == "s0me"); }
       }
     }
+
+    AND_GIVEN("a StringView wrapping a buffer with sufficient capacity for C++ string") {
+      openassetio::Str storage;
+      storage.reserve(expectedStr.size());
+
+      OPENASSETIO_NS(StringView) actualStringView{storage.capacity(), storage.data(), 0};
+
+      WHEN("assignStringView is used to copy the C++ string to the StringView") {
+        openassetio::assignStringView(&actualStringView, expectedStr);
+
+        THEN("StringView matches source string") { CHECK(actualStringView == expectedStr); }
```

Comment:
[fix] Should we check also in these cases that it has copied and detached from the src string?

Reply:
Updated tests to ensure data was copied not just pointed to in 61fcc94f